screeps-profiler.js

```diff
@@ -116,15 +132,23 @@ const Profiler = {
     Game.notify(Profiler.output());
   },
 
-  output() {
+  output(numresults) {
+    if(!numresults) {
+      numresults = 20
+    }
+
+    if(!Memory.profiler || !Memory.profiler.enabledTick) {
+      return 'Profiler not active.'
+    }
+
     const elapsedTicks = Game.time - Memory.profiler.enabledTick + 1;
     const header = 'calls\t\ttime\t\tavg\t\tfunction';
     const footer = [
       `Avg: ${(Memory.profiler.totalTime / elapsedTicks).toFixed(2)}`,
       `Total: ${Memory.profiler.totalTime.toFixed(2)}`,
       `Ticks: ${elapsedTicks}`,
     ].join('\t');
-    return [].concat(header, Profiler.lines().slice(0, 20), footer).join('\n');
+    return [].concat(header, Profiler.lines().slice(0, numresults), footer).join('\n');
```

Comment:
This is ok for now, but I've been thinking about changing the way this works. For anything other than email (email needs to be limited), log the full amount.